### DISCUSSION TITLE
Installing 4.15.2 - cannot complete, error in SQL syntax

### DISCUSSION THREAD

**Discussion by bcinteractive** [timschofield/webERP#275] [Installation Issues] [answered]
Let the app do the install wizard. (I did not create a config.php file.)
Here is the error message after input all install info:

> PHP Fatal error:  Uncaught mysqli_sql_exception: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'SET FOREIGN_KEY_CHECKS = 0' at line 1 in /home/doobie/webdocs/webERP/install/index.php:1224

The SQL statement was:  `SET names UTF8; SET FOREIGN_KEY_CHECKS = 0`

I run this via phpmyadmin and it does not send any error message.

I cannot complete installation.  Please advice.

Using php 8.3.x, mariadb 10.11.6

**Comment by timschofield** [comment, marked as answer]
4.15.2 is quite old. We are moving quickly towards a v5.0 release. If you are looking at it for testing I would use the code from the master branch. It has a completely rewritten installer. It is feature complete, but there are a number of bugs still and that branch is quite fast moving.

However, this appears to be the mysqli driver baulking at having the two statements thrown at it in the one line - no idea why. If you change that line and replace it with:

					$SQL = "SET names UTF8";
					$result = mysqli_query($db,$SQL);
					$SQL = "SET FOREIGN_KEY_CHECKS = 0";
					$result = mysqli_query($db,$SQL);

then it appears to work fine for me here.

Thanks for your report,
Tim

**Reply by bcinteractive** [reply]
Thanks, Tim.

I feel your excitement re v5.  But ok, I'll dive into v5 later.

Thanks again.

**Comment by dalers** [comment]
@bcinteractive v4.15.2.zip on PHP 7 is a pretty stable point in time if that's what you're looking for, but it has some issues and there may be other divergences from current sversions of MariaDb and/or MySQL.

However, if you require PHP 8[1] or if you want 4.15.2 to be fixed and you can manage with stabilization pains, you should follow the master branch and drink from the fire hydrant (fire plug?) 

Cheers,
Dale

[1] I need PHP8 because several other PHP apps I host already require php8 - MantisBT, Moodle, WackoWiki and SuiteCRM (SuiteCRM should be php8-only soon).

**Comment by bcinteractive** [comment]
Just checking if I'm on the right 'master' branch.  I've managed to download and install this version -- but it is showing version 4.15.1
![Screenshot from 2024-12-06 12-54-07](https://github.com/user-attachments/assets/981cb1fe-7291-43c1-930d-aa90f22884d0)

**Comment by dalers** [comment]
Something doesn't seem right, I would have expected you to see 4.15.2, or 5.0.0.rc.9 if you downloaded within the last 8 hours. 

How did you download? How did you create the database? Are you familiar with git? (git would be the best way to download and update the code).

**Reply by bcinteractive** [reply]
I used the download zip link to download the app.
![Screenshot from 2024-12-06 16-11-07](https://github.com/user-attachments/assets/b91348aa-107c-4a7d-a092-6c40f755779f)

Database, I let run the install wizard when first accessing webERP in the browser.  Very small familiarity with git (used only locally).

**Reply by bcinteractive** [reply]
Wait a minute.  I've just installed gh and looks like I'm able to dowload the app with git.

**Comment by dalers** [comment]
I don't recognise the desktop. What OS are you using? How did you use git locally before? (command line?)

The [Development page on weberp.org](https://weberp.org/Development.html) is a good overview of clong the project repository, except in your case I would clone the project repo directly and not create my own Github fork (you can change the process later if you want to contribute a bugfix or feature back to the project).

Instead of using git, you can download a zip file with the current state of the master branch from GitHub. 

![image](https://github.com/user-attachments/assets/179cb1fe-8d65-4b93-afa9-82e0230f29dd)

**Reply by bcinteractive** [reply]
Hi,

OS: debian 12, gnome desktop
Using git locally for my own exercise apps - nothing uploaded --- just some exercises, I guess.

Thank you very much for the screenshot with notes -- very vivid, did I mention clear? :smiley:  :smiley:  :smiley: 

Right now, I'm gonna stay away from master and work with the stable version -- need to get familiar with my webERP workflow first and make a working app, then I'll try to catch up with the v5 -- very exciting indeed.

Thanks for this cheat sheet, @dalers .

**Comment by timschofield** [comment]
The only thing with 4.15.2 is that it is not PHP8 compatible. If you
are happy with that then it is fine.

Thanks for your post,
Tim


On Sat, 7 Dec 2024 at 03:49, bcinteractive ***@***.***> wrote:
>
> Hi,
>
> OS: debian 12, gnome desktop
> Using git locally for my own exercise apps - nothing uploaded --- just some exercises, I guess.
>
> Thank you very much for the screenshot with notes -- very vivid, did I mention clear? 😃 😃 😃
>
> Right now, I'm gonna stay away from master and work with the stable version -- need to get familiar with my webERP workflow first and make a working app, then I'll try to catch up with the v5 -- very exciting indeed.
>
> Thanks for this cheat sheet, @dalers .
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.Message ID: ***@***.***>



-- 
www.weberp.org
@TimSchofield2
Blog: https://kwamoja.home.blog/
